### PR TITLE
Add staggered tree and block generation

### LINKED ISSUE #3: Generate trees with staggered tree spacing
Stagger tree spacing such that each row is not on the same level as all the other rows. The stagger distance to be added is half the tree to tree distance along the row.

### PR DESCRIPTION
**Pull request by PatilVrush**

### CODE REVIEW

**Comment by shrijitsingh99** [comment]
@PatilVrush Can you please close the previous PR. Let's also try to get this merged in today, in time for the Monday sync.

File: data/pre_gen_info.txt
Comment (shrijitsingh99) [inline data/pre_gen_info.txt]: What's the red color for?

File: data/pre_gen_info.txt
Comment (shrijitsingh99) [inline data/pre_gen_info.txt:23]: Please remove extra new line

File: scripts/create_farm.py
Comment (shrijitsingh99) [inline scripts/create_farm.py]: Remove extra new lines. Only one is needed at end of file.

File: scripts/get_plane_model.py
Comment (shrijitsingh99) [inline scripts/get_plane_model.py]:
```suggestion
          <pose>{:.5f} {:.5f} 0 0 0 0</pose>
```
This should contain 6 numbers corresponding to x y z r p y

File: scripts/get_plane_model.py
Comment (shrijitsingh99) [inline scripts/get_plane_model.py:13]:
```suggestion
              <size>{:.5f} {:.5f} 0</size>
```
I believe this should also contain another number indicating z. Correct me if I am wrong.

File: scripts/old_create_farm.py
Comment (shrijitsingh99) [inline scripts/old_create_farm.py]: Please delete this, git already tracks changes. We don't need to store old scripts. We can look at the git history to see older version.

File: worlds/spraybot.world
Comment (shrijitsingh99) [inline worlds/spraybot.world]: What the red color for?

File: worlds/spraybot.world
Comment (shrijitsingh99) [inline worlds/spraybot.world]: I think this is for all `<pose>x y z r p y<pose>`. The pose tags should contain 6 values everywhere, it contains only 5 in every instance. This could potentially lead to errors in future Gazebo versions. Please change it, to add the missing values in all instances.

File: worlds/spraybot.world
Comment (shrijitsingh99) [inline worlds/spraybot.world:33]: 2 points:
1. Name doesn't need to be in quotes. Please remove it.
2. Could you add some comments regarding what the naming scheme is:
Eg. `tree_{block number}_{row number}_{column/tree number}`.

File: worlds/spraybot.world
Comment (shrijitsingh99) [inline worlds/spraybot.world]: Can you add a screenshot of the world so that I can get a better idea of what it looks like.

File: scripts/create_farm.py
Comment (shrijitsingh99) [inline scripts/create_farm.py:92]: While I won't push for it. Previously
```
Generate the string sections for all of the trees in a row. Previously each function had doctoring like shown below, which explained what the function did, what parameters it took as input and what it returned.
Args:
----
y_ind (int): the index of the row
y_pos (float): the y position of the row
num_trees_row (int): the num of trees in the row
```

The lack of documentation will make it hard for anyone to modify this script.
I would be fine if you skipped documentation for now, but created an issue for it to add at some point.
@simi-asher You can chime in with your thoughts

**Comment by shrijitsingh99** [comment]
@PatilVrush One important point to add besides the comments. Please delete the `__pycache__/*.pyc` files. Make sure you don't push any binary files.

I did't comment on this when the script was initially added since the script was quite small and did a small task, but now it seems to be more complex and has a growing feature set. So it would be better to have a main function, with helper functions or even possibly reflector everything into a class. Open to making this into an open issue and deferring for later due to time constraints, but something definitely to be noted.
@simi-asher Pinging if you have any additional comments.

File: scripts/create_farm.py
Comment (simi-asher) [inline scripts/create_farm.py]: Move this comment above to include the spacing variables too since they are also user set

**Comment by simi-asher** [comment]
> @PatilVrush One important point to add besides the comments. Please delete the `__pycache__/*.pyc` files. Make sure you don't push any binary files.
> 
> I did't comment on this when the script was initially added since the script was quite small and did a small task, but now it seems to be more complex and has a growing feature set. So it would be better to have a main function, with helper functions or even possibly reflector everything into a class. Open to making this into an open issue and deferring for later due to time constraints, but something definitely to be noted. @simi-asher Pinging if you have any additional comments.

I agree that refactor to class would help readability but I think for now we don't have major updates to sim and the created world file can be used as is with the README (@PatilVrush update the README if this is not the case) instructions. We can make this an issue but make it low priority. For now, readability can be improved by adding some more description about the logic/algorithm being used to generate blocks and I would approve the PR after that.

**Comment by simi-asher** [comment]
Closed by accident, sorry

File: data/pre_gen_info.txt
Comment (PatilVrush) [reply data/pre_gen_info.txt]: Changed my bad. Anyways it doesn't reflect in simulation.

File: data/pre_gen_info.txt
Comment (PatilVrush) [reply data/pre_gen_info.txt:23]: Done

File: scripts/create_farm.py
Comment (PatilVrush) [reply scripts/create_farm.py]: Done

File: scripts/get_plane_model.py
Comment (PatilVrush) [reply scripts/get_plane_model.py]: Corrected

File: scripts/create_farm.py
Comment (PatilVrush) [reply scripts/create_farm.py]: Done

File: scripts/create_farm.py
Comment (shrijitsingh99) [inline scripts/create_farm.py]: Please remove extra lines

File: scripts/get_plane_model.py
Comment (shrijitsingh99) [inline scripts/get_plane_model.py:13]: Still missing a value here. There should be 3 values

File: worlds/spraybot.world
Comment (shrijitsingh99) [reply worlds/spraybot.world:33]: @PatilVrush Could you resolve point 2

**Comment by shrijitsingh99** [comment]
Added comments regarding some remaining minor fixes. Most of it looks good to go.
After the remaining fixes, please create an issue regarding the remaining documentation work. Once the issue is up, will merge this in.

**Review by shrijitsingh99 — APPROVED** [review]

**Comment by shrijitsingh99** [comment]
Merged. Awesome work @PatilVrush!

**Pull request merged**
